### ISSUE TITLE
[Enhancement]: copy permissions to subfolders

### ISSUE THREAD

**Issue by dgingold** [dnnsoftware/Dnn.Platform#5904]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description of problem

the old file manager could copy permissions down to subfolders. This is a sorely missed feature of Resource Manager. My client said they'd never have moved to DNN if they knew the many subfolders of files they had would need to be manually modified one by one. 

### Description of solution

provide a checkbox in Resource Manager folder edit > permissions to copy current folder's permissions to all subfolders, meaning recursively through all subfolders below the current folder. This functionality needs to work for both files local to the site (ie, in the /portals/<portalID> folders and offsite folders via providers like Azure Connector and Evotiva Global Storage (since my specific issue has its files in offsite storage)

### Description of alternatives considered

Is there a SQL statement that will accomplish this as a stop-gap measure? I have 50GB of files that have been moved from another site as we migrated that site into a section of an existing DNN site. Can the old file manager be reinstalled (frankly I'd rather do a SQL statement than to add and then untangle the file manager after running the permission copy)

### Do you be plan to contribute code for this enhancement?

- [ ] Yes

### Would you be interested in sponsoring this enhancement?

- [X] Yes

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by 6TELOIV** [comment]
For implementation, the API (partially) exists already through `DotNetNuke.Services.FileSystem.FolderManager.CopyParentFolderPermissions` and would just need to recursively be called on children folders of the selected folder.

**Comment by jeremy-farrance** [comment]
Yes, we've run in to this also and our client said a very similar thing. And his comments got even worse when he discovered modifying permissions on existing folders doesn't work either (#5857). Luckily this was months ago and he is back in the happiness zone. I hope after DNN v10 comes out, there will be time to circle back to Resource Manage for a little TLC.
